### PR TITLE
Add Audit Logs to the Python SDK

### PR DESCRIPTION
**Pull request by jonatascastro12**
Add AuditLogs calls to the SDK:
-  `create_event`
-  `create_export`
-  `get_export`

Also made a few tweaks to expose `code` and `errors` payload in exceptions.

### CODE REVIEW

File: workos/audit_logs.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                    event[actor][id] (str)
+                    event[actor][name] (str)
+                    event[actor][type] (str)
+                    event[actor][metadata] (dict)
+                event[targets] (list[dict]) - List of event targets
+                event[context] (dict) - Attributes of event context
+                    event[context][location] (str)
+                    event[context][user_agent] (str)
+                event[metadata] (dict) - Extra metadata
+            idempotency_key (str) - Optional idempotency key
```
Comment (jonatascastro12) [inline workos/audit_logs.py:45]: @robframpton Could take a look at the descriptions of the fields on this docstring?

File: workos/audit_logs.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    ):
+        """Trigger the creation of an export of audit logs.
+
+        Args:
+            organization (str) - Organization's unique identifier
+            range_start (str) - Start date of the range filter
+            range_end (str) - End date of the range filter
+            actions (list) - Optional list of actions to filter
+            actors (list) - Optional list of actors to filter
+            targets (list) - Optional list of targets to filter
```
Comment (jonatascastro12) [inline workos/audit_logs.py:83]: Here as well. @robframpton

File: workos/__about__.py [hunk 3]
```diff
@@ -12,7 +12,7 @@
 
 __package_url__ = "https://github.com/workos-inc/workos-python"
 
-__version__ = "1.14.0"
+__version__ = "1.15.0"
```
Comment (robframpton) [inline workos/__about__.py:15]: We typically bump the versions in a dedicated branch after feature changes have been merged in.

File: workos/audit_logs.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        return self._request_helper
+
+    def create_event(self, organization, event, idempotency_key=None):
+        """Create an Audit Logs event.
+
+        Args:
+            organization (str) - Organization's unique identifier
+            event (dict) - An event object
+                event[action] (string) - The event action
+                event[version] (int) - The version of event
```
Comment (robframpton) [inline workos/audit_logs.py:33]:
```suggestion
                event[version] (int) - The schema version of the event
```

File: workos/audit_logs.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        Args:
+            organization (str) - Organization's unique identifier
+            range_start (str) - Start date of the range filter
+            range_end (str) - End date of the range filter
+            actions (list) - Optional list of actions to filter
+            actors (list) - Optional list of actors to filter
+            targets (list) - Optional list of targets to filter
+
+        Returns:
+            dict: Object that describes the exported audit logs job
```
Comment (robframpton) [inline workos/audit_logs.py:86]:
```suggestion
            dict: Object that describes the audit log export
```

File: workos/audit_logs.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+            token=workos.api_key,
+        )
+
+        return WorkOSAuditLogsExport.construct_from_response(response)
+
+    def get_export(self, export_id):
+        """Retrieve an created export.
+
+        Returns:
+            dict: Object that describes the exported event
```
Comment (robframpton) [inline workos/audit_logs.py:117]:
```suggestion
            dict: Object that describes the audit log export
```

File: workos/resources/audit_logs_export.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
+from workos.resources.base import WorkOSBaseResource
+from workos.resources.event_action import WorkOSEventAction
+
+
+class WorkOSAuditLogsExport(WorkOSBaseResource):
```
Comment (robframpton) [inline workos/resources/audit_logs_export.py:5]: This is what we've been calling the entity elsewhere.
```suggestion
class WorkOSAuditLogExport(WorkOSBaseResource):
```

File: workos/audit_logs.py [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+        actions=None,
+        actors=None,
+        targets=None,
+    ):
+        """Trigger the creation of an export of audit logs.
+
+        Args:
+            organization (str) - Organization's unique identifier
+            range_start (str) - Start date of the range filter
+            range_end (str) - End date of the range filter
```
Comment (robframpton) [inline workos/audit_logs.py:80]:
```suggestion
            range_start (str) - Start date of the date range filter
            range_end (str) - End date of the date range filter
```

File: workos/audit_logs.py [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+        response = self.request_helper.request(
+            EVENTS_PATH,
+            method=REQUEST_METHOD_POST,
+            params=payload,
+            headers=headers,
+            token=workos.api_key,
+        )
+
+        return response["success"]
```
Comment (robframpton) [inline workos/audit_logs.py:64]: Not sure if it's possible in Python, but we've been returning `Void` in the other SDKs as the `success` property is kind of a meaningless value that we return from various endpoints. There is some more context in this Slack thread that lead to this: https://work-os.slack.com/archives/C037Z7GMMAB/p1657562404232559

**Comment by robframpton** [comment]
It looks like we'll need to add some additional error mapping here: https://github.com/workos/workos-python/blob/main/workos/utils/request.py#L87-L96 similarly to what I did for the Kotlin SDK: https://github.com/workos/workos-kotlin/pull/127/files#diff-31e1355895fca164d2b3ce61cd7fc1d1bfbf83c304893e6727fc010a67ad19dcR252-R255

Essentially we just want to make sure the `code` and `errors` properties are propagated.

File: workos/audit_logs.py (hunk 9 above)
Comment (jonatascastro12) [reply workos/audit_logs.py:64]: So `void` in Python means `return None`. Nothing wrong with that, but I was trying to be consistent with other places in the Python SDK. 

https://github.com/workos/workos-python/blob/78f705670cfb76e8c9888e45a1148fea34ac0763/workos/passwordless.py#L70
https://github.com/workos/workos-python/blob/d5171cad529142b2470b5c69db47516776ac9b41/workos/audit_trail.py#L73

What do you think?

By the way, we don't have Python typing in our SDK (I want to propose that).

Comment (robframpton) [reply workos/audit_logs.py]: Yeah it's going against historical convention in the other SDKs as well, I think we'll still want to make the change though to keep the various `createEvent` implementations the same.

**Pull request merged**